src/screens/sync/index.tsx

```diff
@@ -28,21 +33,38 @@ const styles = StyleSheet.create({
 
 export const SyncScreen: ScreenComponent<"Sync"> = ({ route }) => {
   const [viewMode, setViewMode] = React.useState<ViewMode>("list");
+  const [modalContent, setModalContent] = React.useState<TitleAndDescription>({
+    title: "",
+    description: "",
+  });
+  const { sheetRef, openSheet } = useBottomSheetModal({ openOnMount: false });
+
+  function setAndOpenModal(titleAndDescription: TitleAndDescription) {
+    setModalContent(titleAndDescription);
+    openSheet();
+  }
 
   return (
-    <ScrollView
-      style={styles.container}
-      contentContainerStyle={styles.scrollContentContainer}
-    >
-      <ProjectInfo
-        name="Project Catapult"
-        viewMode={viewMode}
-        toggleViewMode={() => {
-          setViewMode((prev) => (prev === "list" ? "bubbles" : "list"));
-        }}
-      />
-      <Devices mode={viewMode} />
-    </ScrollView>
+    <React.Fragment>
+      <ScrollView
+        style={styles.container}
+        contentContainerStyle={styles.scrollContentContainer}
+      >
+        <ProjectInfo
+          name="Project Catapult"
+          viewMode={viewMode}
+          toggleViewMode={() => {
+            setViewMode((prev) => (prev === "list" ? "bubbles" : "list"));
+          }}
+        />
+        <Devices>
+          {viewMode === "list" ? (
+            <DeviceList setAndOpenModal={setAndOpenModal} />
+          ) : null}
+        </Devices>
+      </ScrollView>
+      <SyncGroupBottomSheet content={modalContent} ref={sheetRef} />
```

Comment:
generally think it's easier to keep the BottomSheetModal usage co-located with the useBottomSheetModal hook usage, and only dynamically updating the content within the sheet. basically extracting the BottomSheetModal out of SyncGroupBottomSheet such that here it looks like:

```suggestion
      <BottomSheetModal>
          <SyncGroupContent />
      </BottomSheetModal>
```

not sure if there's any consequence to using multiple bottom sheet modals in the same componeent, but this alternate approach ensures that only one exists for this screen, and only the content within it changes based on state

Reply:
https://github.com/digidem/mapeo-mobile-shell/pull/28/commits/f63808067cd2927413d7e868415aa0a2fcafec8c